### PR TITLE
[Tizen] Add material renderers for Editor and Picker

### PR DESCRIPTION
**Pull request by shyunMin**
### Description of Change ###

This PR includes two commits. The one is to add material renderers for `Editor`, `Picker`, `DatePicker`, `TimePicker`, and the other one is to fix the renderes for picker to hide a cursor.

### Issues Resolved ### 

 None

### API Changes ###

The APIs in Tizen backend has been changed as follows:
 (`Xamarin.Forms.Platform.Tizen` and `Xamarin.Forms.Material.Tizen`)

Added:
  - Entry
    protected virtual OnTextBlockFocused();
    protected virtual OnTextBlockUnfocused();
    protected virtual OnEntryLayoutFocused();
    protected virtual OnEntryLayoutUnfocused(); 

  - IEntry
    event EventHandler TextBlockFocused
    event EventHandler TextBlockUnfocused
    event EventHandler EntryLayoutFocused
    event EventHandler EntryLayoutUnfocused
	
  - EditorRenderer
    protected virtual EEntry CreateNativeControl()

  - TimePickerRenderer
    protected virtual EEntry CreateNativeControl()

  - DatePickerRenderer
    protected virtual EEntry CreateNativeControl()
	
  - PickerRenderer
    protected virtual EEntry CreateNativeControl()

  - TimePickerRenderer
    protected TimeSpan Time

  - MEditor : MaterialEntry
  - MPicker : MEditor
  
  - MaterialEntryRenderer : EntryRenderer
  - MaterialPckerRenderer : PickerRenderer
  - MaterialDatePckerRenderer : DatePickerRenderer
  - MaterialTimePickerRenderer : TimePickerRenderer

Changed :
  - EditorRenderer
    void UpdateTextColor() => protected virtual void UpdateTextColor()

  - PickerRenderer
    void UpdateTitleColor() => protected virtual void UpdateTitleColor()
	
  - DatePickerRenderer
    void OnDateTimeChanged() => protected virtual void OnDateTimeChanged()
    void UpdateDate() => protected virtual void UpdateDate()
    void UpdateTextColor() => protected virtual void UpdateTextColor()

  - TimePickerRenderer
    void UpdateTextColor() => protected virtual void UpdateTextColor()
    void UpdateTimeAndFormat() =>protected virtual void UpdateTimeAndFormat()

### Platforms Affected ### 

- Tizen

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

- Before
editor_before_gif
picker_before

- After
editor_after_gif
picker_after

- Fixed `Picker` to hide a cursor on TV
  - Before
tv_picker_before
  - After
tv_picker_after

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Tizen/Native/Entry.cs [hunk 1]
```diff
@@ -364,6 +385,26 @@ public virtual ESize Measure(int availableWidth, int availableHeight)
 
 		}
 
+		protected virtual void InvokeTextBlockFocused()
```
Comment (rookiejava) [inline Xamarin.Forms.Platform.Tizen/Native/Entry.cs]: Is there any special reason to provide it as a method?

Comment (shyunMin) [reply Xamarin.Forms.Platform.Tizen/Native/Entry.cs]: This method will be called in MEditor to trigger TextBlockFocused event, and it is needed for implementing MaterialPickerRenderer.

**Comment by shyunMin** [comment]
I've updated a description of API changes

File: Xamarin.Forms.Platform.Tizen/Native/Entry.cs (hunk 1 above)
Comment (rookiejava) [reply Xamarin.Forms.Platform.Tizen/Native/Entry.cs]: why don't you invoke those events declared in `IEntry` instead?

**Comment by rookiejava** [comment]
> I've updated a description of API changes

I think there are something still missing in material projects.

File: Xamarin.Forms.Platform.Tizen/Native/Entry.cs (hunk 1 above)
Comment (rookiejava) [reply Xamarin.Forms.Platform.Tizen/Native/Entry.cs]: consider to apply [this pattern](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-raise-base-class-events-in-derived-classes)

Comment (shyunMin) [reply Xamarin.Forms.Platform.Tizen/Native/Entry.cs]: Fixed.

**Comment by shyunMin** [comment]
Rebased. I've changed the base branch from master to 4.5.0.

**Comment by samhouts** [comment]
@shyunMin Sorry this one sat too long! I've retargetted it to 4.6.0, but there are some conflicts now. Would you be able to resolve those? Thanks!

**Comment by shyunMin** [comment]
@samhouts  Thank you for notify me. I resolved conflicts.

**Pull request merged**